src/progress/progress_set.rs

```diff
@@ -63,6 +67,22 @@ impl Configuration {
             learners: learners.into_iter().collect(),
         }
     }
+
+    /// Create a new `ConfState` from the configuration itself.
+    pub fn to_conf_state(&self) -> ConfState {
+        let mut state = ConfState::default();
+        state.set_nodes(self.voters.iter().cloned().collect());
```

Comment:
I suggest you to use `vec` instead of `HashSet`. Benches shows that iterating vec is way faster than querying HashSet when the size of vec is very small.

```
running 2 tests
test bench_set_contains ... bench:          73 ns/iter (+/- 0)
test bench_vec_contains ... bench:          11 ns/iter (+/- 0)
```

Reply:
Let's do it after the PR.